### PR TITLE
fix: Add existential operators to properties before calling split in processBrowserPreRequest

### LINKED ISSUE #28950: SIGABRT: Cannot read properties of undefined (reading 'split') | i.processBrowserPreRequest
### Current behavior

I ran `cypress open` with [this repo](https://github.com/freon27/cypress-test-tiny) (`npm i` and `cypress open`), then clicked E2E tests and Chrome v121. Upon opening the browser, Cypress crashed with the error below. It did not happen the second time I opened Cypress.

```
Cannot read properties of undefined (reading 'split')
TypeError: Cannot read properties of undefined (reading 'split')
    at <embedded>:4432:32336
    at Map.forEach (<anonymous>)
    at i.processBrowserPreRequest (<embedded>:4432:32254)
    at N.shouldIgnorePendingRequest (<embedded>:4432:42079)
    at N.addPendingBrowserPreRequest (<embedded>:4432:41117)
    at d (<embedded>:4432:40190)
    at _ (<embedded>:4432:36011)
    at _.emit (node:events:513:28)
    at f (<embedded>:4491:55225)
    at onEnd (<embedded>:4491:55769)
    at q (<embedded>:4491:53135)
    at N (<embedded>:4491:53761)
    at ee.<anonymous> (<embedded>:4491:55678)
    at ee.emit (node:events:525:35)
    at ee.onRequestError (<embedded>:2050:100575)
    at ClientRequest.emit (node:events:525:35)
    at Socket.socketErrorListener (node:_http_client:502:9)
    at Socket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
GET /__cypress/runner/fonts/fa-solid-900.woff2 200 1.204 ms - 126828
The Test Runner unexpectedly exited via a exit event with signal SIGABRT

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------

Platform: darwin-x64 (23.2.0)
Cypress Version: 13.6.4
```

This seems to be failing within this function within our code, I'm unsure if it would be with the `documentUrl`

https://github.com/cypress-io/cypress/blob/renovate/cross-fetch-3.x/packages/proxy/lib/http/util/service-worker-manager.ts#L121

Or the `url`

https://github.com/cypress-io/cypress/blob/renovate/cross-fetch-3.x/packages/proxy/lib/http/util/service-worker-manager.ts#L133

The other split calls have an existence operator to check before calling split. 

### Desired behavior

_No response_

### Test code to reproduce

[this repo](https://github.com/freon27/cypress-test-tiny) (`npm i` and `cypress open`) - although I've been unable to reproduce this again.

### Cypress Version

13.6.4

### Node version

18

### Operating System

Mac

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- Closes https://github.com/cypress-io/cypress/issues/28950

### Additional details

Seems this was introduced in https://github.com/cypress-io/cypress/pull/28517

If these properties don't exist, they should no longer crash. It may result in null values for some of the properties we're trying to check to process service worker requests, but this seems better than outright crashing. Perhaps there's a better long term solution to ensure these exist before getting these values.

### Steps to test

Good question. I have no idea why this would even be the case and don't see a clear way to unit test it.

### How has the user experience changed?

Users should not see cypress crash with:

```
Cannot read properties of undefined (reading 'split')
TypeError: Cannot read properties of undefined (reading 'split')
    at <embedded>:4432:32336
    at Map.forEach (<anonymous>)
    at i.processBrowserPreRequest (<embedded>:4432:32254)
    at N.shouldIgnorePendingRequest (<embedded>:4432:42079)
    at N.addPendingBrowserPreRequest (<embedded>:4432:41117)
    at d (<embedded>:4432:40190)
    at _ (<embedded>:4432:36011)
    at _.emit (node:events:513:28)
    at f (<embedded>:4491:55225)
    at onEnd (<embedded>:4491:55769)
    at q (<embedded>:4491:53135)
    at N (<embedded>:4491:53761)
    at ee.<anonymous> (<embedded>:4491:55678)
    at ee.emit (node:events:525:35)
    at ee.onRequestError (<embedded>:2050:100575)
    at ClientRequest.emit (node:events:525:35)
    at Socket.socketErrorListener (node:_http_client:502:9)
    at Socket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
GET /__cypress/runner/fonts/fa-solid-900.woff2 200 1.204 ms - 126828
The Test Runner unexpectedly exited via a exit event with signal SIGABRT
```

### PR Tasks

- [ ] Have tests been added/updated?
- [N/A] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [N/A] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by emilyrohrbough — APPROVED** [review]

**Comment by ryanthemanuel** [comment]
This is frustrating because according to the docs those are both required fields.  Hmmm.

**Review by mschile — APPROVED** [review]

File: packages/proxy/lib/http/util/service-worker-manager.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ export class ServiceWorkerManager {
 
     this.serviceWorkerRegistrations.forEach((registration) => {
       const activatedServiceWorker = registration.activatedServiceWorker
-      const paramlessDocumentURL = browserPreRequest.documentURL.split('?')[0]
+      const paramlessDocumentURL = browserPreRequest.documentURL?.split('?')[0]
```
Comment (ryanthemanuel) [inline packages/proxy/lib/http/util/service-worker-manager.ts]:
```suggestion
      const paramlessDocumentURL = browserPreRequest.documentURL?.split('?')[0] || ''
```

File: packages/proxy/lib/http/util/service-worker-manager.ts [hunk 2]
```diff
@@ -130,7 +130,7 @@ export class ServiceWorkerManager {
         return
       }
 
-      const paramlessURL = browserPreRequest.url.split('?')[0]
+      const paramlessURL = browserPreRequest.url?.split('?')[0]
```
Comment (ryanthemanuel) [inline packages/proxy/lib/http/util/service-worker-manager.ts]:
```suggestion
      const paramlessURL = browserPreRequest.url?.split('?')[0] || ''
```

**Comment by ryanthemanuel** [comment]
I added the comments defaulting to '' as I think if they would have also crashed a couple of lines later. Looks good now though.

**Review by ryanthemanuel — APPROVED** [review]

**Comment by MikeMcC399** [comment]
@This PR is already merged, however the changelog link is incorrect:

```text
Fixed a regression introduced in [`13.6.3`](https://docs.cypress.io/guides/references/changelog#13.6.3) where Cypress could crash when processing service worker requests through our proxy. Fixes [#28950](https://github.com/cypress-io/cypress/issues/28950).
```

It should be
```text
[`13.6.3`](https://docs.cypress.io/guides/references/changelog#13-6-3)
```
if [`13.6.3`](https://docs.cypress.io/guides/references/changelog#13-6-3) is going to work correctly.

- See PR https://github.com/cypress-io/cypress/pull/28957 for fix

**Comment by jennifer-shehane** [comment]
@MikeMcC399 Thanks Mike!

**Pull request merged**